### PR TITLE
fix: clean release tags and release notes

### PR DESCRIPTION
**Pull request by osama-salman99**
* [x] Code is up-to-date with the `main` branch.
* [x] You've successfully built and run the tests locally.
* [ ] There are new or updated unit tests validating the change.

### :pencil: Description
- Clean release tags and release notes

### :link: Related Issues
N/A

### CODE REVIEW

File: clean-release-tags.sh
Comment (osama-salman99) [inline clean-release-tags.sh:1]: Can be re-programmed to delete all existing tags with "vX.X.X" format. What do you think?

File: .github/workflows/generator-draft-release.yaml [hunk 1]
```diff
@@ -17,3 +34,6 @@ jobs:
         run: npx semantic-release
         env:
           GITHUB_TOKEN: ${{ secrets.GITHUB_TOKEN }}
+      - name: Clean release tags
+        if: always()
+        run: ./clean-release-tags.sh
```
Comment (osama-salman99) [inline .github/workflows/generator-draft-release.yaml:41]: We can run this step at the start of the workflow to cleanup any previous mistakenly-created tags. What are your thoughts?

Comment (anssari1) [reply .github/workflows/generator-draft-release.yaml:41]: Do it!

Comment (osama-salman99) [reply .github/workflows/generator-draft-release.yaml:41]: Done ✅

**Review by anssari1 — APPROVED** [review]

**Pull request merged**